### ISSUE TITLE
_app rendering twice on dev env

### ISSUE THREAD

**Issue by himadrinath** [vercel/next.js#36920]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 14.17.4
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.1.7-canary.6
      react: 18.1.0
      react-dom: 18.1.0
```

### Describe the Bug

when i run the project in development env _app page called twice but in production env working fine.

### Expected Behavior

it should call only once

### To Reproduce

here is the codesandbox link to reproduce the error

https://codesandbox.io/s/virtual-scroller-infinite-scroll-forked-hthpe0?file=/pages/_app.js

**Comment by SukkaW** [comment]
Because you have enabled `reactStrictMode` in your `next.config.js`: https://reactjs.org/docs/strict-mode.html

> Strict mode checks are run in development mode only; they do not impact the production build.
> ...
> Strict mode can’t automatically detect side effects for you, but it can help you spot them by making them a little more deterministic. This is done by intentionally double-invoking the following functions:
>
> -Class component constructor, render, and shouldComponentUpdate methods
> - Class component static getDerivedStateFromProps method
> - Function component bodies
> - State updater functions (the first argument to setState)
> - Functions passed to useState, useMemo, or useReducer

You should either disable React Strict Mode or fix your code to make your component works even if React renders it twice.

**Comment by himadrinath** [comment]
@SukkaW  it worked perfectly well before in react 17 and next@12.1.5 this problem started after i updated to react 18 and next@12.1.6
then i don't understand then how come it worked before

**Comment by balazsorban44** [comment]
This is working as intended, the explanation is at: https://github.com/vercel/next.js/issues/36920#issuecomment-1126967498

This is also a duplicate of https://github.com/vercel/next.js/issues/35822
